### PR TITLE
Feature/error flow

### PR DESCRIPTION
**Pull request by tduong992**
This PR solves:
- #131 Clearly define Error Flow
- replace user role with user username

### CODE REVIEW

**Comment by tduong992** [comment]
English:
Screenshot 2019-05-06 20 58 07

German:
Screenshot 2019-05-06 20 57 57

**Comment by kernda02** [comment]
The error message "User not found" is also displayed with a correct user but wrong password. 
Should we rename "User not found" to "invalid credentials"?

File: public/locales/de/translation.json [hunk 1]
```diff
@@ -10,13 +10,16 @@
...
+        "bookmarkSubject": "Fach bookmarken",
+        "userRole": "Ihre Rolle ist:"
     },
     "login": {
         "message": "Auf LaMaS anmelden",
         "password": "Passwort",
         "email": "E-Mail Adresse",
-        "login": "Anmelden"
+        "login": "Anmelden",
+        "errorHeader": "Benutzer nicht gefunden",
```
Comment (kernda02) [inline public/locales/de/translation.json]: change to "Ungültige Kombination Benuzter/Passwort"

File: public/locales/en/translation.json [hunk 2]
```diff
@@ -10,13 +10,16 @@
...
+        "bookmarkSubject": "Bookmark subject",
+        "userRole": "Your role is:"
     },
     "login": {
         "message": "Log-in to LaMaS",
         "password": "password",
         "email": "e-mail address",
-        "login": "Login"
+        "login": "Login",
+        "errorHeader": "User not found",
```
Comment (kernda02) [inline public/locales/en/translation.json]: change to "invalid credentials"

**Review by kernda02 — CHANGES_REQUESTED** [review]
change "user not found" to "invalid credentials"

**Comment by wumacH** [comment]
there are many messages defined we could retrieve from firebase in case of a unsuccessfull login (https://github.com/firebase/firebaseui-web/blob/master/soy/strings.soy), so i just changed the title of the message to a more general version (as you suggested)

**Comment by wumacH** [comment]
ready to merge, thanks for your work :-)

**Pull request merged**